cdc/owner.go

```diff
@@ -696,12 +696,22 @@ func (o *Owner) flushChangeFeedInfos(ctx context.Context) error {
 		}
 	}
 	if time.Since(o.gcSafepointLastUpdate) > GCSafepointUpdateInterval {
-		_, err := o.pdClient.UpdateServiceGCSafePoint(ctx, CDCServiceSafePointID, o.gcTTL, minCheckpointTs)
+		actual, err := o.pdClient.UpdateServiceGCSafePoint(ctx, CDCServiceSafePointID, o.gcTTL, minCheckpointTs)
 		if err != nil {
 			log.Warn("failed to update service safe point", zap.Error(err))
-		} else {
-			o.gcSafepointLastUpdate = time.Now()
+			return cerror.ErrUpdateServiceSafepointFailed.Wrap(err)
```

Comment:
Returning error fails changefeed immediately. Could you fail the check if the last successful update is older than 5 minutes (default GC lifetime is 10 minutes) to prevent changefeed from being disrupted by transient errors?